#!/usr/bin/env python3
PACKAGE = "robot_joy"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_vel_trans",          double_t, 0, "The absolute value of the maximum translational velocity for the robot in m/s", 0.5,  0.0, 2.0)
gen.add("max_vel_th",             double_t, 0, "The absolute value of the maximum rotational velocity for the robot in rad/s",  1.0, 0.0, 3.0)
gen.add("acc_lim_trans",          double_t, 0, "The translational acceleration limit of the robot in meters/sec^2",             0.5, 0.1, 1.0)
gen.add("acc_lim_th",             double_t, 0, "The rotational acceleration limit of the robot in radians/sec^2",               0.2, 0.1, 2)
gen.add("dec_lim_trans",          double_t, 0, "The translational deceleration limit of the robot in meters/sec^2",             0.5, 0.1, 1)
gen.add("dec_lim_th",             double_t, 0, "The rotational deceleration limit of the robot in radians/sec^2",               1.0, 0.1, 2)
gen.add("smooth_transition_time", double_t, 0, "smooth transition time",                                                        1.0, 0.1, 1.0)
gen.add("smooth_control",         bool_t,   0, "joy smooth control",                                                            True)

exit(gen.generate(PACKAGE, "robot_joy_vel_dynamic", "Vel"))